kore/src/Kore/OnePath/Step.hs

```diff
@@ -112,28 +112,31 @@ if x then phi else psi
 with these rules
 
 @
-if true then x else y => x
-if false then phi else psi
+if true  then u else v => u
+if false then u else v => v
 @
 
 there would be two 'RewritePattern's, @phi and x=true@ and @psi and x=false@.
+If only the first rule was present, the results would be a 'RewritePattern' with
+@phi and x=true@ and a 'Remainder' with @psi and not x=true@.
 
 When rewriting the same pattern with an rule that does not match, e.g.
 
 @
 x + y => x +Int y
 @
 
-then the rewrite result should be 'Bottom'.
+then rewriting produces no children.
+
 -}
 data StrategyPattern patt
     = RewritePattern !patt
     -- ^ Pattern on which a normal 'Rewrite' can be applied. Also used
     -- for the start patterns.
-    | Stuck !patt
+    | Remainder !patt
     -- ^ Pattern which can't be rewritten anymore.
-    | Bottom
-    -- ^ special representation for a bottom rewriting/simplification result.
+    | Goal
```

Comment:
This is not always a goal, e.g. while rewriting you can get a state split which cannot be reduced immediately to bottom, but which becomes bottom in the simplification step. You might be replacing a confusing thing with something equally confusing.

Not sure what's the best way to solve this, you might try splitting the state.

Reply:
I can see how this might also be confusing. Maybe it would be better to call this `GoalReached` (which is what I meant) or even just `Done`?

Reply:
I decided to revert this particular change. We can revisit it later if necessary.